ingestors/npm.go

```diff
@@ -1,154 +1,130 @@
+		return sequence, results
+	}
+	defer response.Body.Close()
+
+	body, _ := io.ReadAll(response.Body)
+	sequenceList, _, _, _ := jsonparser.Get(body, "results")
+	_, _ = jsonparser.ArrayEach(sequenceList, func(value []byte, dataType jsonparser.ValueType, offset int, err error) {
+		name, _ := jsonparser.GetString(value, "id")
+		seq, _ := jsonparser.GetInt(value, "seq")
+
+		// The new NPM feed only provides names and sequences, so we don't get Version, CreatedAt or DiscoveryLag.
+		results = append(results,
+			data.PackageVersion{
+				Platform: "npm",
+				Name:     name,
+				Sequence: strconv.FormatInt(seq, 10),
+			})
+	})
+	lastSequence, err := jsonparser.GetInt(body, "last_seq")
+	if err != nil {
+		log.WithFields(log.Fields{"ingestor": ingestor.Name(), "error": err}).Error()
+		lastSequence = sequence
+	}
+
+	return lastSequence, results
 }
 
-func getCouchClient() *kivik.Client {
-	kivikClient, err := kivik.New("couch", NPMRegistryHostname)
+func (ingestor *NPM) getCurrentSequence() int64 {
+	bookmark, err := getBookmark(ingestor, "")
 	if err != nil {
-		log.WithFields(log.Fields{"ingestor": "npm"}).Fatal(err)
+		log.WithFields(log.Fields{"ingestor": ingestor.Name(), "error": err}).Fatal()
+	}
+
+	var currentSequence int64
+	if bookmark != "" {
+		currentSequence, _ = strconv.ParseInt(bookmark, 10, 64)
+	} else if currentSequence == 0 {
+		currentSequence = ingestor.getLatestSequence()
+		log.WithFields(log.Fields{"ingestor": ingestor.Name(), "msg": fmt.Sprintf("No NPM bookmark saved, using latest published sequence %d", currentSequence)}).Info()
 	}
-	return kivikClient
+
+	return currentSequence
 }
 
-// Get a list of querystring params for the _changes endpoint.
-// Docs: https://docs.couchdb.org/en/3.2.0/api/database/changes.html
-func getOptionsForChangesFeed(since string) kivik.Option {
-	return kivik.Params(map[string]interface{}{
-		"feed":         "continuous",
-		"since":        since,
-		"include_docs": true,
-		// NB: previously with "timeout: 60000 * 2", we kept getting an internal error from npm, which surfaced as
-		// "stream error: stream ID 123; INTERNAL_ERROR". They showed up when there was no activity for 50 seconds,
-		// and we're not sure why. But setting a heartbeat ensures the connection stays open every 5 seconds via empty line.
-		"heartbeat": ReconnectRetryDelaySeconds * 1000,
-	})
+// As a fallback, fetch the latest published sequence from https://replicate.npmjs.com/registry/.
+func (ingestor *NPM) getLatestSequence() int64 {
+	response, err := depperGetUrl(npmIndexUrl)
+	if err != nil {
+		log.WithFields(log.Fields{"ingestor": ingestor.Name(), "error": err}).Fatal()
+	}
+	defer response.Body.Close()
+	body, _ := io.ReadAll(response.Body)
+	latestSequence, err := jsonparser.GetInt(body, "update_seq")
+	if err != nil {
+		log.WithFields(log.Fields{"ingestor": ingestor.Name(), "error": err}).Fatal()
+	}
+
+	return latestSequence
```

Comment:
@katzj add a method here to get the "update_seq" from https://registry.npmjs.org/registry, as a fallback value.